### PR TITLE
Add 1-step continuous aggregate policies

### LINKED ISSUE #1476: Add documentation for 1-step policy management for CAggs
Please add the documentation for the newly added feature of 1-step policy for CAggs. Commit details for the same is as follows,
https://github.com/timescale/timescaledb/commit/bca65f46975f61e36e2ce57dc1ae7d80d6cccf05
https://github.com/timescale/timescaledb/commit/16fdb6ca5e3a42c120a45c7dc635e03c12f4b2e9

### PR DESCRIPTION
**Pull request by charislam**
Update the documentation to include the new 1-step continuous aggregate
policies.

Adds:
*   An API reference for every policy
*   ~~A section in the compression, retention, and continuous aggregate
    refresh sections to reference the new features~~

Updates:
*   Updates troubleshooting frontmatter in the new directory to
    reference the new APIs where applicable

# Links

Fixes #1476 

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: api/alter_policies.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+|Name|Type|Description|
+|-|-|-|
+|`relation`|`REGCLASS`|The continuous aggregate that you want to alter policies for|
+
+## Optional arguments
+
+|Name|Type|Description|
+|-|-|-|
+|`if_not_exists`|`BOOL`|When true, prints a warning instead of erroring if the policy doesn't exist. Defaults to false.|
```
Comment (charislam) [inline api/alter_policies.md:45]: @RafiaSabih Is it the policy or the continuous aggregate that the `if_not_exists` applies to? I assumed the policy

File: api/policies.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+<!-- markdownlint-disable-next-line line-length -->
+## timescaledb_experimental.policies <tag type="community" content="Community" /><tag type="experimental" content="Experimental" />
+<!-- vale Google.Headings = YES -->
+
+The `policies` view provides information on all policies set on continuous
+aggregates.
+
+<highlight type="note">
+Only policies applying to continuous aggregates are shown in this view. Policies
+applying to regular hypertables or regular materialized views are not displayed.
```
Comment (charislam) [inline api/policies.md:25]: @RafiaSabih Is this true? This is what I got from reading the comments on the tests

File: api/remove_all_policies.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+|Name|Type|Description|
+|-|-|-|
+|`relation`|`REGCLASS`|The continuous aggregate to remove all policies from|
+
+## Optional arguments
+
+|Name|Type|Description|
+|-|-|-|
+|`if_exists`|`BOOL`|When true, prints a warning instead of erroring if no policies exist. Defaults to false.|
```
Comment (charislam) [inline api/remove_all_policies.md:40]: @RafiaSabih Is it "if _no_ policies exist" or does it also error if any one of the policies doesn't exist?

File: api/remove_continuous_aggregate_policy.md [hunk 4]
```diff
@@ -9,23 +9,39 @@ api:
...
+|`continuous_aggregate`|`REGCLASS`|Name of the continuous aggregate the policy should be removed from|
 
-### Sample usage
+## Optional arguments
 
-Remove the refresh policy from the 'cpu_view' continuous aggregate:
+|Name|Type|Description|
+|-|-|-|
+|`if_not_exists`|`BOOL`|When true, prints a warning instead of erroring if the policy doesn't exist. Defaults to false. Deprecated in TimescaleDB 2.8.|
+|`if_exists`|`BOOL`|When true, prints a warning instead of erroring if the policy doesn't exist. Defaults to `NULL`, which is the same as false. Introduced in TimescaleDB 2.8.|
```
Comment (charislam) [inline api/remove_continuous_aggregate_policy.md:39]: @RafiaSabih Not entirely sure I have this right

File: api/alter_policies.md (hunk 1 above)
Comment (RafiaSabih) [reply api/alter_policies.md:45]: Yes, you are right, it is referring to policy.

File: api/remove_continuous_aggregate_policy.md (hunk 4 above)
Comment (RafiaSabih) [reply api/remove_continuous_aggregate_policy.md:39]: This is just to change the name of the parameter.

File: api/policies.md (hunk 2 above)
Comment (RafiaSabih) [reply api/policies.md:25]: Yes.

**Review by RafiaSabih — APPROVED** [review]

File: _partials/_caggs-one-step-policy.mdx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  Use a one-step policy definition to set a {props.policyType} policy on a
+  continuous aggregate
+</h2>
+
+In TimescaleDB 2.8 and above, policy management on continuous aggregates is
+simplified. You can add, change, or remove the refresh, compression, and data
+retention policies on a continuous aggregate using a one-step API. For more
+information, see the APIs for [adding policies][add-policies], [altering
+policies][alter-policies], and [removing policies][remove-policies]. Note that
+this feature is experimental.
```
Comment (Loquacity) [inline _partials/_caggs-one-step-policy.mdx:11]:
```suggestion
information, see the APIs for [adding policies][add-policies], 
[altering policies][alter-policies], and [removing policies][remove-policies]. 
Note that this feature is experimental.
```

Don't we have an experimental block we can add here?

File: _partials/_caggs-one-step-policy.mdx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+<highlight type="note">
+  The policies changed using this API apply to the continuous aggregate, not to
```
Comment (Loquacity) [inline _partials/_caggs-one-step-policy.mdx:14]:
```suggestion
  When you change policies with this API, the changes apply to the continuous aggregate, not to
```

File: api/remove_all_policies.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+## Returns
+
+Returns true if successful.
+
+## Sample usage
+
+Given a continuous aggregate named `example_continuous_aggregate`, remove all
+policies from it. This includes refresh policies, compression policies, and data
+retention policies. It doesn't include user-defined jobs:
```
Comment (Loquacity) [inline api/remove_all_policies.md:50]:
```suggestion
Remove all policies from a continuous aggregate named 
`example_continuous_aggregate`. This includes refresh policies, compression 
policies, and data retention policies. It doesn't include user-defined jobs:
```

**Review by Loquacity — APPROVED** [review]

File: _partials/_caggs-one-step-policy.mdx [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import Experimental from "versionContent/_partials/_experimental.mdx";
+
+<h2>
+  Use a one-step policy definition to set a {props.policyType} policy on a
```
Comment (jfjoly) [inline _partials/_caggs-one-step-policy.mdx:4]: Where will this be shown?
Until we get more feedback, we are not trying to move users to the new functions vs. the current process.

Comment (charislam) [reply _partials/_caggs-one-step-policy.mdx:4]: I put it on all the how-to guides for setting refresh, compression, and retention policies. If you don't want to advertise the new functions too broadly, I can take it off the how-to guides for now and just add the API references.

**Comment by Loquacity** [comment]
@jfjoly Just waiting on a response from you so we can merge this. Please let us know if you need any changes.

**Comment by jfjoly** [comment]
@charislam and I discussed that we will just add the API references. I'm not sure the PR has been updated.

**Pull request merged**